modules/curl/8.7.1.bcr.1/patches/add_bazel_support.patch

```diff
@@ -0,0 +1,644 @@
++    name = "http_only_setting",
++    flag_values = {":http_only": "true"},
++)
++
++expand_template(
++    name = "curl_config",
++    out = "lib/curl_config.h",
++    substitutions =
++        get_substitutions() |
++        select({
++            ":use_mbedtls_setting": dict([enabled_setting_substitution("USE_MBEDTLS")]),
++            "//conditions:default": dict([disabled_setting_substitution("USE_MBEDTLS")]),
++        }),
++    template = "lib/curl_config.h.cmake",
++)
++
++cc_library(
++    name = "curl",
++    srcs = [
++        ":curl_config",
++    ] + glob([
++        "lib/**/*.c",
++        "lib/**/*.h",
++    ]),
++    hdrs = glob(["include/curl/*.h"]),
++    defines = [
++        "CURL_STATICLIB",
++    ],
++    includes = [
++        "include",
++        "lib",
++    ],
++    linkopts = select({
++        "@platforms//os:windows": [
++            "-DEFAULTLIB:ws2_32.lib",
++            "-DEFAULTLIB:advapi32.lib",
++            "-DEFAULTLIB:crypt32.lib",
++            "-DEFAULTLIB:Normaliz.lib",
++        ],
++        "@platforms//os:macos": [
++            "-Wl,-framework,SystemConfiguration",
++            "-lpthread",
++        ],
++        "//conditions:default": [
++            "-lpthread",
++        ],
++    }),
++    local_defines = [
++        "BUILDING_LIBCURL",
++        "HAVE_CONFIG_H",
++    ] + select({
++        ":http_only_setting": ["HTTP_ONLY"],
++        "//conditions:default": [],
++    }) + select({
++        "@platforms//os:windows": [
++            "WIN32",
++        ],
++        "//conditions:default": [
++            "CURL_HIDDEN_SYMBOLS",
++        ],
++    }),
++    visibility = ["//visibility:public"],
++    deps = select({
++        ":use_mbedtls_setting": ["@mbedtls"],
++        "//conditions:default": [],
++    }),
++)
++
++cc_binary(
++    name = "linktest",
++    linkopts = select({
++        "@platforms//os:macos": ["-all_load"],
++        "//conditions:default": [],
++    }),
++    linkshared = True,
++    visibility = ["//visibility:private"],
++    deps = [":curl"],
++)
+diff --git a/MODULE.bazel b/MODULE.bazel
+new file mode 100644
+index 000000000..8991a973a
+--- /dev/null
++++ b/MODULE.bazel
+@@ -0,0 +1,9 @@
++module(
++    name = "curl",
++    version = "8.7.1.bcr.1",
++    compatibility_level = 0,
++)
++
++bazel_dep(name = "bazel_skylib", version = "1.5.0")
++bazel_dep(name = "mbedtls", version = "3.6.0")
++bazel_dep(name = "platforms", version = "0.0.8")
+diff --git a/config.bzl b/config.bzl
+new file mode 100644
+index 000000000..3b95ad364
+--- /dev/null
++++ b/config.bzl
+@@ -0,0 +1,509 @@
++"""Generate platform specific template substitutions for curl_config.h.cmake"""
++
++load("@bazel_skylib//lib:new_sets.bzl", "sets")
++
++_ENABLED_SETTINGS = sets.make([
++    "CURL_DISABLE_LDAP",
++    "CURL_DISABLE_OPENSSL_AUTO_LOAD_CONFIG",
++    "ENABLE_IPV6",
++    "HAVE_ALARM",
++    "HAVE_ARC4RANDOM",
++    "HAVE_ARPA_INET_H",
++    "HAVE_ATOMIC",
++    "HAVE_BASENAME",
++    "HAVE_BOOL_T",
```

Comment:
I think the default curl config for windows does a better job with some of these defines 
https://github.com/curl/curl/blob/master/lib/config-win32.h#L113-L115

This patch breaks our project in compared to just using the default config.

```
ERROR: C:/users/zekew/_bazel_zekew/xrgzj2nk/external/ecsact_cli~/ecsact/cli/BUILD.bazel:28:10: Compiling ecsact/cli/ecsact_cli.cc failed: (Exit 2): cl.exe failed: 
error executing CppCompile command (from target @@ecsact_cli~//ecsact/cli:ecsact) C:\Program Files\Microsoft Visual Studio\2022\Community\VC\Tools\MSVC\14.39.33519\bin\HostX64\x64\cl.exe ... (remaining 1 argument skipped)
C:\Users\zekew\_bazel_zekew\xrgzj2nk\execroot\_main\external\curl~\lib\curl_setup_once.h(242): error C2628: '<unnamed-enum-bool_false>' followed by 'bool' is illegal (did you forget a ';'?)
C:\Users\zekew\_bazel_zekew\xrgzj2nk\execroot\_main\external\curl~\lib\curl_setup_once.h(242): warning C4091: 'typedef ': ignored on left of '<unnamed-enum-bool_false>' when no variable is declared

```